cloud/services/internalloadbalancers/internalloadbalancers.go

```diff
@@ -171,7 +172,8 @@ func (s *Service) getAvailablePrivateIP(ctx context.Context, resourceGroup, vnet
 		ip = azure.DefaultInternalLBIPAddress
 		if subnetCIDR != infrav1.DefaultControlPlaneSubnetCIDR {
 			// If the user provided a custom subnet CIDR without providing a private IP, try finding an available IP in the subnet space
-			ip = subnetCIDR[0:7] + "0"
+			index := strings.LastIndex(subnetCIDR, ".")
+			ip = subnetCIDR[0:(index+1)] + "0"
```

Comment:
Great catch @dklyle! Could you please add a unit test to validate the fix (ie. that fails without the fix and passes with the fix)?